ydb/core/blobstorage/dsproxy/dsproxy_get_block.cpp

```
@@ -0,0 +1,139 @@
+    }
+
+    std::unique_ptr<IEventBase> RestartQuery(ui32 counter) override {
+        ++*Mon->NodeMon->RestartGetBlock;
+        auto ev = std::make_unique<TEvBlobStorage::TEvGetBlock>(Deadline, TabletId);
+        ev->RestartCounter = counter;
+        return ev;
+    }
+public:
+    ::NMonitoring::TDynamicCounters::TCounterPtr& GetActiveCounter() const override {
+        return Mon->ActiveGetBlock;
+    }
+
+    ERequestType GetRequestType() const override {
+        return ERequestType::GetBlock;
+    }
+
+    TBlobStorageGroupGetBlockRequest(TBlobStorageGroupGetBlockParameters& params)
+        : TBlobStorageGroupRequestActor(params)
+        , TabletId(params.Common.Event->TabletId)
+        , Deadline(params.Common.Event->Deadline)
+        , Requests(0)
+        , Responses(0)
+        , QuorumTracker(Info.Get())
+    {}
+
+    void Bootstrap() override {
+        DSP_LOG_INFO_S("DSPGB04", "bootstrap"
+            << " ActorId# " << SelfId()
+            << " Group# " << Info->GroupID
+            << " Deadline# " << Deadline
+            << " RestartCounter# " << RestartCounter);
+
+        for (const auto& vdisk : Info->GetVDisks()) {
+            const ui64 cookie = TVDiskIdShort(Info->GetVDiskId(vdisk.OrderNumber)).GetRaw();
+
+            auto vd = Info->GetVDiskId(vdisk.OrderNumber);
+            DSP_LOG_DEBUG_S("DSGB05", "Sending TEvVGetBlock"
+                << " vDiskId# " << vd
+                << " node# " << Info->GetActorId(vd).NodeId());
+
+            auto msg = std::make_unique<TEvBlobStorage::TEvVGetBlock>(TabletId, vd, Deadline);
+            SendToQueue(std::move(msg), cookie);
+            ++Requests;
+        }
+
+        Become(&TBlobStorageGroupGetBlockRequest::StateWait);
+
+        if (Requests == 0) {
+            ReplyAndDie(NKikimrProto::OK);
+        }
+    }
+
+    STATEFN(StateWait) {
+        if (ProcessEvent(ev)) {
+            return;
+        }
+        switch (ev->GetTypeRewrite()) {
```

and in other cases? don't just silently drop unknown messages, we should fail